### ISSUE TITLE
Create Dynamic Loading Assembly Helper

### ISSUE THREAD

**Issue by james-h-stephens** [Abantech/Efficio#27]
Create assembly loader to load device assemblies based on configuration.  We are seeking to move to this structure for the C++ code:
![whatsapp image 2016-04-09 at 19 47 06](https://cloud.githubusercontent.com/assets/23285467/20781065/1a6e9008-b74e-11e6-83f8-5831f8649fb7.jpeg)

**Comment by james-h-stephens** [comment]
@joeabsoftdev ever done this before?

**Comment by james-h-stephens** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Create Dynamic Loading Assembly Helper](https://trello.com/c/zbSmWIk6/1-create-dynamic-loading-assembly-helper)

**Comment by joeabsoftdev** [comment]
@james-h-stephens i don't think so, but would want to initialize immediately post your instructions.

**Comment by james-h-stephens** [comment]
@joeabsoftdev Basically, we want to be able to search a filesystem area and find any DLLs of interest. From those DLLs, we want to be able to instantiate some classes

**Comment by joeabsoftdev** [comment]
If structures are acceptable, especially for kernel operations, should we write in C?

**Comment by joeabsoftdev** [comment]
@james-h-stephens If it will support assembly loading, I would be keen to utilize assembly language and/or machine code.

**Comment by joeabsoftdev** [comment]
This looks like the right struct: https://msdn.microsoft.com/en-us/library/aa365740(v=vs.85).aspx. I've only seen it in C++
